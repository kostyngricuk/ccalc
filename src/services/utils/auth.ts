import { TUser } from "../../types/user";

export const hasAdditionalInfo = (user: TUser) => {
  if (
    !user?.gender ||
    !user?.age ||
    !user?.height ||
    !user?.weight ||
    !user?.weightGoal
  ) {
    return false;
  }
  return true;
};