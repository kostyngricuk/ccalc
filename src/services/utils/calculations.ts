import { ISelectedProduct } from '@services/types/products';

export const getKkal = ({
  proto,
  carbo,
  fats
}: {
  proto: number,
  carbo: number,
  fats: number
}) => proto * 4 + carbo * 4 + fats * 9;

export interface ITottalValues {
  weight: number,
  kkal: number,
  proto: number,
  carbo: number,
  fats: number
}

export const getTottal = (items: ISelectedProduct[]): ITottalValues => {
  const weight = items.reduce((acc, item) => acc + item.weight, 0);
  const kkal = items.reduce((acc, item) => acc + item.kkal, 0);
  const proto = items.reduce((acc, item) => acc + item.proto, 0);
  const carbo = items.reduce((acc, item) => acc + item.carbo, 0);
  const fats = items.reduce((acc, item) => acc + item.fats, 0);
  return {
    weight,
    kkal,
    proto,
    carbo,
    fats
  }
}

export const getNutritionByWeight = ({
  product,
  newWeight
}: {
  product: ISelectedProduct,
  newWeight: number
}) => {
  const {
    kkal,
    proto,
    fats,
    carbo,
    weight
  } = product;

  const validWeight = newWeight || 1;

  return {
    kkal: kkal / weight * validWeight,
    proto: proto / weight * validWeight,
    fats: fats / weight * validWeight,
    carbo: carbo / weight * validWeight,
    weight: validWeight
  };;
}
