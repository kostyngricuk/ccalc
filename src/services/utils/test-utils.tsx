/* eslint-disable import/no-extraneous-dependencies */
import React, {ReactNode} from 'react'
import lightTheme from '@services/styled/themes'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { RootState, setupStore } from '@services/store';
import { Genders, TUser } from '@services/types/user';
import { BrowserRouter } from 'react-router-dom';

export function Providers({
  children,
  state,
}: {
  children: ReactNode,
  state: Partial<RootState>,
}) {
  const store = setupStore(state);
  return (
    <ThemeProvider theme={lightTheme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>
  )
}

export const defaultMockUser: TUser = {
  id: 'unicStringId',
  height: 180,
  weight: 85,
  weightGoal: 80,
  age: 26,
  gender: Genders.man,
  email: 'mockUser@gmail.com',
  calorieWidget: {
    limit: 2300,
    eaten: 1980,
  },
}
