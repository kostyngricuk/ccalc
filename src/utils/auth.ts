import { TUser } from 'types/user';

const hasAdditionalInfo = (user: TUser) => {
  if ( user?.gender && user?.age && user?.height && user?.weight && user?.weightGoal) {
    return true;
  }
  return false;
};
export default hasAdditionalInfo;