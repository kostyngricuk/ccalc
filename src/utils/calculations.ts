import { ISelectedProduct } from 'types/products';

export const getKkal = ({
  proto,
  carbo,
  fats
}: {
  proto: number,
  carbo: number,
  fats: number
}) => proto * 4 + carbo * 4 + fats * 9;

export interface ITottalValues {
  weight: number,
  kkal: number,
  proto: number,
  carbo: number,
  fats: number
}

export const getTottal = (items: ISelectedProduct[]): ITottalValues => (
  items.reduce((acc, item) => (
    {
      weight: acc.weight + item.weight,
      kkal: acc.kkal + item.kkal,
      proto: acc.proto + item.proto,
      carbo: acc.carbo + item.carbo,
      fats: acc.fats + item.fats
    }
  ), { weight: 0, kkal: 0, proto: 0, carbo: 0, fats: 0 })
)

export const getNutritionByWeight = ({
  product,
  newWeight = 100
}: {
  product: ISelectedProduct,
  newWeight?: number
}) => {
  const {
    kkal,
    proto,
    fats,
    carbo,
    weight
  } = product;

  return {
    kkal: kkal / weight * newWeight,
    proto: proto / weight * newWeight,
    fats: fats / weight * newWeight,
    carbo: carbo / weight * newWeight,
    weight: newWeight
  };
}
