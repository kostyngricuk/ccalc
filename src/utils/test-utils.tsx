/* eslint-disable import/no-extraneous-dependencies */
import React, {PropsWithChildren, ReactNode} from 'react'
import lightTheme from 'styles/themes'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { RootState } from 'store/store';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import configureMockStore, { MockStore, MockStoreCreator } from 'redux-mock-store';
import { I18nextProvider } from 'react-i18next';
import i18n from 'locales/i18nForTests';
import { Genders, TUser } from 'types/user';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>
}

export function Providers({
  children,
  store,
}: {
  children: ReactNode,
  store: MockStore,
}) {
  return (
    <ThemeProvider theme={lightTheme}>
      <Provider store={store}>
        <I18nextProvider i18n={i18n} defaultNS="translation">
          <BrowserRouter>
            {children}
          </BrowserRouter>
        </I18nextProvider>
      </Provider>
    </ThemeProvider>
  )
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  const mockStore: MockStoreCreator = configureMockStore();
  const store: MockStore = mockStore(preloadedState);
  return {
    store,
    ...render(
      ui,
      {
        wrapper: ({ children }: PropsWithChildren<{}>) => (
          <Providers store={store}>
            { children}
          </Providers>
        ),
        ...renderOptions
      }
    )
  }
}

// re-export everything
export * from '@testing-library/react'

// override render method
export {renderWithProviders as render}

export const defaultMockUser: TUser = {
  id: 'unicStringId',
  height: 180,
  weight: 85,
  weightGoal: 80,
  age: 26,
  gender: Genders.man,
  email: 'mockUser@gmail.com',
  calorieWidget: {
    limit: 2300,
    eaten: 1980,
  },
}
