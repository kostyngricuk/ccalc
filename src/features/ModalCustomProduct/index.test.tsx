import React from "react";
import { defaultMockProduct, defaultMockSelectedProduct, fireEvent, render, waitFor } from "utils/test-utils";
import ModalAddCustomProduct from "features/ModalCustomProduct";

describe('Test ModalCustomProduct component', () => {
  const mockToggleModal = jest.fn();
  const state = {
    product: {
      items: [defaultMockProduct],
      selectedItems: [defaultMockSelectedProduct]
    }
  }

  it('Render component', () => {
    const wrapper = render(<ModalAddCustomProduct toggleModal={mockToggleModal}/>, {
      preloadedState: state
    });
    const form = wrapper.getByTestId('form');
    expect(form).toBeInTheDocument();
  })

  it('Change value for input', async () => {
    const wrapper = render(<ModalAddCustomProduct toggleModal={mockToggleModal}/>, {
      preloadedState: state
    });
    const form = wrapper.getByTestId('form');
    expect(form).toBeInTheDocument();

    const protoInput = wrapper.getByLabelText("Proteins");

    fireEvent.input(protoInput, {
      target: {
        value: 200
      }
    });

    expect(protoInput).toHaveValue(200);
  })
})