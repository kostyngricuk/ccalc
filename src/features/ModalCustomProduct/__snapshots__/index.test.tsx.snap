// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ModalCustomProduct component Should make snapshot 1`] = `
<form
  class="sc-eqUzNf iaUwNn"
  data-testid="form"
>
  <div
    class="sc-kAycRU ihKhcg Input"
  >
    <label
      class="sc-imWZod fNnZtJ is-required"
    >
      <input
        autocomplete="off"
        data-testid="input"
        name="name"
        required=""
        type="text"
        value=""
      />
      <span
        class="label"
      >
        Product name
      </span>
    </label>
  </div>
  <div
    class="sc-iGgVNO RWVnE"
  >
    <div
      class="sc-kAycRU ihKhcg Input"
    >
      <label
        class="sc-imWZod fNnZtJ is-required"
      >
        <input
          autocomplete="off"
          data-testid="input"
          name="proto"
          required=""
          type="number"
          value=""
        />
        <span
          class="label"
        >
          Proteins
        </span>
      </label>
    </div>
    <div
      class="sc-kAycRU ihKhcg Input"
    >
      <label
        class="sc-imWZod fNnZtJ is-required"
      >
        <input
          autocomplete="off"
          data-testid="input"
          name="fats"
          required=""
          type="number"
          value=""
        />
        <span
          class="label"
        >
          Fats
        </span>
      </label>
    </div>
    <div
      class="sc-kAycRU ihKhcg Input"
    >
      <label
        class="sc-imWZod fNnZtJ is-required"
      >
        <input
          autocomplete="off"
          data-testid="input"
          name="carbo"
          required=""
          type="number"
          value=""
        />
        <span
          class="label"
        >
          Carbohydrates
        </span>
      </label>
    </div>
  </div>
  <div
    class="sc-gsFSjX dZxRnx"
  >
    <button
      aria-label=""
      class="sc-gEvDqW ddrchV"
      color="third"
      type="button"
    >
      Cancel
    </button>
    <button
      aria-label=""
      class="sc-gEvDqW bjpgsz"
      color="primary"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
